libunwind/src/libunwind_ext.h

```diff
@@ -12,6 +12,7 @@
 #ifndef __LIBUNWIND_EXT__
 #define __LIBUNWIND_EXT__
 
+#ifndef __wasm__
```

Comment:
This feels like the wrong way to do this. We don't typically just if-def out a header for a particular platform.

Maybe track down the users of this header, and make them work under WASM?

Reply:
The problem is that __WASM_EXCEPTIONS__ macro is not defined by clang, so using any other the header without __wasm__ will trigger ODR violation. Remote unwinding is not a thing for wasm, so i do not see what's wrong here

Reply:
I would much rather you guard the usage of this header in each file.

That way it's clear to the reader in context that the include is unused under a certain configuration.
If the entire file contains no WASM code, then it probably shouldn't mention WASM at all.

Can you please explain what the `__WASM_EXCEPTIONS__` or `__USING_WASM_EXCEPTIONS__` was?

Reply:
wasm has its own exception handling mechanism. i do not know why __USING_WASM_EXCEPTION__ is not defined, because the author of the code did not change the upstream code correctly i guess.